### ISSUE TITLE
More accessible custom headers.

### ISSUE THREAD

**Issue by DylanPiercey** [form-data/form-data#133]
With https://github.com/felixge/node-form-data/pull/128 we can easily modify the code to automatically wrap a "header" object into the full header for convenience, 

This would change this:
```javascript
var CRLF = '\r\n';
var form = new FormData();

var options = {
  header: CRLF + '--' + form.getBoundary() + CRLF + 'X-Custom-Header: 123' + CRLF + CRLF,
  knownLength: 1
};

form.append('my_buffer', buffer, options);

form.submit('http://example.com/', function(err, res) {
  if (err) throw err;
  console.log('Done');
});
```

Into this:
```javascript
var form = new FormData();

var options = {
  header: {
     'X-Custom-Header': '123'
  }
  knownLength: 1
};

form.append('my_buffer', buffer, options);

form.submit('http://example.com/', function(err, res) {
  if (err) throw err;
  console.log('Done');
});
```

Which is a bit more user friendly with less manual string manipulation.

**Comment by alexindigo** [comment]
Totally, if we preserve original way, you know for people who looooove control. :)

**Comment by alexindigo** [comment]
@DylanPiercey Looks like this is the last one for 1.0, and we'll be good to go.

**Comment by DylanPiercey** [comment]
@alexindigo sweet, I'll try to get to this within the next few days.

**Comment by alexindigo** [comment]
Thanks man.

**Comment by alexindigo** [comment]
Fixed in #190
